<a href="https://colab.research.google.com/github/Mafred02/Programacion/blob/main/Parcial2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Parcial II

Nombre: Nicolás Mafred Rodríguez Durán

Codigo: 2260676

Parte II
Introduccion 1


In [2]:
import numpy as np

In [259]:
np.random.seed(0)
datos = np.random.randint(177, 197, 50)
datos


array([189, 192, 177, 180, 180, 184, 186, 196, 195, 181, 183, 189, 178,
       183, 184, 191, 194, 182, 190, 185, 186, 196, 193, 196, 182, 192,
       192, 177, 195, 180, 194, 196, 196, 196, 191, 184, 177, 178, 186,
       177, 187, 180, 188, 195, 179, 177, 177, 181, 182, 183])

In [282]:
#1)
datos[datos < 185]

array([177, 180, 180, 184, 181, 183, 178, 183, 184, 182, 182, 177, 180,
       184, 177, 178, 177, 180, 179, 177, 177, 181, 182, 183])

In [283]:
#2)
import pandas as pd

df = pd.DataFrame(datos)
normalizar = lambda x: (x - x.mean()) / x.std()
resultado = normalizar(df)

print(resultado)

           0
0   0.415785
1   0.867725
2  -1.391976
3  -0.940035
4  -0.940035
5  -0.337449
6  -0.036155
7   1.470312
8   1.319665
9  -0.789389
10 -0.488095
11  0.415785
12 -1.241329
13 -0.488095
14 -0.337449
15  0.717078
16  1.169018
17 -0.638742
18  0.566432
19 -0.186802
20 -0.036155
21  1.470312
22  1.018372
23  1.470312
24 -0.638742
25  0.867725
26  0.867725
27 -1.391976
28  1.319665
29 -0.940035
30  1.169018
31  1.470312
32  1.470312
33  1.470312
34  0.717078
35 -0.337449
36 -1.391976
37 -1.241329
38 -0.036155
39 -1.391976
40  0.114491
41 -0.940035
42  0.265138
43  1.319665
44 -1.090682
45 -1.391976
46 -1.391976
47 -0.789389
48 -0.638742
49 -0.488095


In [284]:
#4)
df = pd.DataFrame(datos)
df

,0
0,189
1,192
2,177
3,180
4,180
5,184
6,186
7,196
8,195
9,181


In [285]:
#3)
df.describe()

,0
count,50.000000
mean,186.240000
std,6.638047
min,177.000000
25%,180.250000
50%,185.500000
75%,192.000000
max,196.000000


Como se puede observar, los datos generados tienen las siguientes caracteristicas: Son un total de 50 datos, su promedio es aproximadamente de 186, su desviacion estandar de 6.6, el valor mas pequeño es un 177 y el maximo 196. Luego, aproximadamente, los resultados del 25% son hasta 180, los resultados de la mitad(50%) hasta el 186 y los del 75% hasta 192.

In [286]:
#5)*
from google.colab import files
import csv

with open("resultados.csv", "w", newline="") as f:
  writer = csv.writer(f)
  writer.writerow(datos)

files.download("resultados.csv")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

Introduccion 2

In [287]:
m = {
    "nombre" : ["Nicolás", "Marian", "Sarai", "Juan", "Galindo" ],
    "edad" : [17, 18, 25, 20, 19],
    "carrera" : ["Matematicas", "Derecho", "Artes", "Fisica", "Matematicas"],
    "ciudad" : ["Bucaramanga", "Bucaramanga", "Medellin", "Cali", "Bogota"]
}
df1 = pd.DataFrame(m)

In [288]:
#1)
df1.head(2)

,nombre,edad,carrera,ciudad
0,Nicolás,17,Matematicas,Bucaramanga
1,Marian,18,Derecho,Bucaramanga


In [289]:
#2)
df1["nombre"]


,nombre
0,Nicolás
1,Marian
2,Sarai
3,Juan
4,Galindo


In [290]:
#3)
df1[df1["edad"] < 20]

,nombre,edad,carrera,ciudad
0,Nicolás,17,Matematicas,Bucaramanga
1,Marian,18,Derecho,Bucaramanga
4,Galindo,19,Matematicas,Bogota


In [291]:
#4)
df1["edad en 10 años"] = df1["edad"].apply(lambda x: x + 10)
df1

,nombre,edad,carrera,ciudad,edad en 10 años
0,Nicolás,17,Matematicas,Bucaramanga,27
1,Marian,18,Derecho,Bucaramanga,28
2,Sarai,25,Artes,Medellin,35
3,Juan,20,Fisica,Cali,30
4,Galindo,19,Matematicas,Bogota,29


In [292]:
#5)
df1.groupby("carrera")["edad"].mean()

,edad
carrera,
Artes,25.0
Derecho,18.0
Fisica,20.0
Matematicas,18.0


In [293]:
#6)*
from google.colab import files
import csv

with open("datos_procesados.csv", "w", newline="") as f:
  writer = csv.writer(f)
  writer.writerow([df1])

files.download("datos_procesados.csv")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

Parte III

Proyecto

In [294]:
#1)
r = np.random.normal(loc=7.2, scale=0.4 , size=52)
d = np.random.gamma(shape=306, scale=0.0229, size=52)


In [295]:
#2)
df2 = pd.DataFrame({
    "prueba_A": r,
    "prueba_B": d
})
df2

,prueba_A,prueba_B
0,6.983633,6.784859
1,6.670709,6.333592
2,7.154880,7.111254
3,7.562938,7.320708
4,7.526108,7.156553
5,7.291639,6.983259
6,6.789528,7.714492
7,7.391010,6.657830
8,7.717079,7.034548
9,6.907417,7.184043


In [143]:
#3)
df2.describe()

,prueba_A,prueba_B
count,52.000000,52.000000
mean,7.136113,7.042446
std,0.441436,0.393833
min,6.042552,6.210692
25%,6.747464,6.775582
50%,7.172289,7.022065
75%,7.450080,7.349698
max,7.980228,7.954398


En la prueba A se obtuvo un promedio aproximado de 7.14, una desviacion de 0.44, una nota minima de 6 y una maxima de 7.98. El 25% saco hasta 6.75, el 50% saco hasta un 7.17 y el 75% saco hasta 7.45.



En la prueba B se obtuvo un promedio aproximado de 7, una desviacion del 0.4 (aproximadamente), un minimo de 6.21 y un maximo de 7.95. El 25% saco hasta un 6.78, el 50% hasta un 7 y el 75% hasta un 7.35.



Con esto podemos concluir que, aunque la diferencia entre los datos obtenidos no es mucha, a los estudiantes en promedio les fue mejor en la Prueba A que en la Prueba B.

In [166]:
#4)
from scipy import stats
stats.shapiro(df2)

ShapiroResult(statistic=np.float64(0.990602270261708), pvalue=np.float64(0.6883444892646673))

Como el valor de p es mayor a 0.05, no se rechaza el supuesto de normalidad con una confianza del 95%.

In [168]:
#5)
p_val = stats.ttest_rel(df2["prueba_A"], df2["prueba_B"])
p_val

TtestResult(statistic=np.float64(1.2957215205070158), pvalue=np.float64(0.20090528976956654), df=np.int64(51))

Como el valor de p es mayor a 0.05, no se rechaza que el puntaje medio en ambas encuentas sea igual con una confianza del 95%.

In [169]:
#6)
stats.kendalltau(df2["prueba_A"], df2["prueba_B"])

SignificanceResult(statistic=np.float64(0.19457013574660634), pvalue=np.float64(0.04176062980916009))

Como el valor de p es menor a 0.05, si se rechaza con una confianza del 95% que los puntajes de ambas encuestas no estan correlacionados.

In [170]:
#7)
stats.spearmanr(df2["prueba_A"], df2["prueba_B"])

SignificanceResult(statistic=np.float64(0.27482284640997184), pvalue=np.float64(0.04863829710679687))

Como el valor de p es menor a 0.05, si se rechaza con una confianza del 95% que los puntajes de ambas encuestas no estan correlacionados.

In [296]:
8#)
import os
os.makedirs("comparacion", exist_ok=True)

In [318]:
#9)
%%writefile comparacion/pruebat.py
from scipy import stats
def prueba_t(df):
  x = df["Prueba_A"]
  p_val = stats.ttest_rel(x)
  return p_val

Overwriting comparacion/pruebat.py


In [307]:
%%writefile comparacion/cor.py
from scipy import stats
def cor(df["Prueba_A"], df["Prueba_B"]):
  p_val = stats.spearmanr(df["Prueba_A"], df["Prueba_B"])
  return p_val

Overwriting comparacion/cor.py


In [320]:
from comparacion import pruebat

import importlib
importlib.reload(pruebat)

df = pd.DataFrame({
"Estudiante": range(1, 11),
"Prueba_A": [7.1, 7.5, 6.8, 7.3, 7.0, 7.4, 6.9, 7.2, 7.6, 7.1],
"Prueba_B": [6.9, 7.2, 7.0, 7.1, 6.8, 7.3, 7.0, 7.1, 7.4, 7.0]
})

pruebat.prueba_t(df)


TypeError: ttest_rel() missing 1 required positional argument: 'b'